libraries/Microsoft.Bot.Builder.Ai.QnA/QnAMaker.cs

```diff
@@ -84,18 +90,19 @@ public QnAMaker(QnAMakerEndpoint endpoint, QnAMakerOptions options = null, HttpC
         /// <summary>
         /// Generates an answer from the knowledge base.
         /// </summary>
-        /// <param name="question">The user question to be queried against your knowledge base.</param>
+        /// <param name="context">The Turn Context that contains the user question to be queried against your knowledge base.</param>
         /// <returns>A list of answers for the user query, sorted in decreasing order of ranking score.</returns>
-        public async Task<QueryResult[]> GetAnswersAsync(string question)
+        public async Task<QueryResult[]> GetAnswersAsync(ITurnContext context)
         {
             var requestUrl = $"{_endpoint.Host}/knowledgebases/{_endpoint.KnowledgeBaseId}/generateanswer";
 
             var request = new HttpRequestMessage(HttpMethod.Post, requestUrl);
+            var messageActivity = context?.Activity?.AsMessageActivity();
 
             var jsonRequest = JsonConvert.SerializeObject(
                 new
```

Comment:
Do we want to leave this as an Anonymous type? Doing so means we have no tests / regressions around it. I would much prefer this to be a strong type with actual Unit tests.